File: examples/python/advanced/serialization/dataclasses.py
from dataclasses import dataclass, asdict
from aws_durable_execution_sdk_python import DurableContext, durable_execution

@dataclass
class Order:
    order_id: str
    amount: float
    customer: str

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    order = Order(
        order_id="ORD-123",
        amount=99.99,
        customer="Jane Doe"
    )
    
    # Convert to dict before passing to step
    result = context.step(process_order, asdict(order))
    return result

File: examples/java/advanced/serialization/dataclasses.java
// Java record (Java 16+) — serializes automatically with Jackson
public record Order(String orderId, double amount, String customer) {}

// Usage in handler
var order = new Order("ORD-123", 99.99, "Jane Doe");
var result = ctx.step("process-order", Map.class, stepCtx -> processOrder(order));
